### ISSUE TITLE
compilation fails:  pf_ring 6.4.1 + kernel-4.7

### ISSUE THREAD

**Issue by tsarangi** [ntop/PF_RING#128]
```
  CC [M]  /builddir/build/BUILD/PF_RING-6.4.1/kernel/pf_ring.o
/builddir/build/BUILD/PF_RING-6.4.1/kernel/pf_ring.c: In function 'ring_setsockopt':
/builddir/build/BUILD/PF_RING-6.4.1/kernel/pf_ring.c:6949: error: implicit declaration of function '__sk_attach_filter'
/builddir/build/BUILD/PF_RING-6.4.1/kernel/pf_ring.c:6963: error: implicit declaration of function '__sk_detach_filter'
make[1]: *** [/builddir/build/BUILD/PF_RING-6.4.1/kernel/pf_ring.o] Error 1
```

**Comment by cardigliano** [comment]
Fixed, please check and reopen if it does not work for you

**Comment by Blzut3** [comment]
This change appears to have broken pfring stable on upstream LTS kernel 4.4.12.

**Comment by cardigliano** [comment]
@Blzut3 what OS? Please provide uname -a and cat /etc/_release_

**Comment by Blzut3** [comment]
@cardigliano Sorry for the late response, I missed the notification.

We're using CentOS 7.2 with the kernel updated to 4.4.12 from kernel.org.

uname -a
`Linux 4.4.12.x86_64 #1 SMP Fri Jun 3 17:55:52 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux`

/etc/release doesn't exist, but redhat-release is:
`CentOS Linux release 7.2.1511 (Core)`

We noticed an issue with sk_attach_filter missing and saw this changeset removed the check for kernel 4.4.8+ and replaced it with only a check for Ubuntu.  Seems like the more correct check would be LINUX_VERSION_CODE >= 4.4.8 && LINUX_VERSION_CODE < 4.7.  I checked the latest 4.4.19 code and it still only exports __sk_attach_filter.
